src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeManager.Templates.cs

```diff
@@ -309,6 +309,10 @@ static bool allValidTypeArguments(bool useUpdatedEscapeRules, AnonymousTypeDescr
 
             static bool hasDefaultScope(bool useUpdatedEscapeRules, AnonymousTypeField field)
             {
+                if (field.HasUnscopedRefAttribute)
+                {
+                    return false;
```

Comment:
>return false; [](http://example.com/codeflow?start=20&length=13)

Have we considered synthesizing generic templates similarly to what we did with default parameter values recently? For example, synthesizing delegate like `delegate void <>f__AnonymousDelegate0<T>([UnscopedRef] ref T x)`.

Reply:
`[UnscopedRef]` parameters should be uncommon, so perhaps we should hold off making that change for now.